tests/test_utils.py

```diff
@@ -130,3 +130,18 @@ def test_new_type_supertype():
     new_new_type = NewType('new_new_type', new_type)
     assert new_type_supertype(new_type) == str
     assert new_type_supertype(new_new_type) == str
+
+
+def test_deep_update_extra_keys():
+    mapping = {'key': {'inner_key': 0}}
+    assert deep_update(mapping, {'other_key': 1}) == {'key': {'inner_key': 0}, 'other_key': 1}
+
+
+def test_deep_update_overwrites_keys():
+    mapping = {'key': {'inner_key': 0}, 'other_key': 1}
+    assert deep_update(mapping, {'key': [1, 2, 3]}) == {'key': [1, 2, 3], 'other_key': 1}
+
+
+def test_deep_update_merges_keys():
+    mapping = {'key': {'inner_key': 0}}
+    assert deep_update(mapping, {'key': {'other_key': 1}}) == {'key': {'inner_key': 0, 'other_key': 1}}
```

Comment:
could you add a test for one level lower to demonstrate that `deep_update` this works recursively.

Reply:
I think `test_deep_update_merges_keys` already tests exactly that, because `{'inner_key': 0}` and `{'other_key': 1}` are themselves values, and they do get merged. Do you want to test one level deeper that that?

Reply:
I see what you mean, but yes, one level deeper would be good I think.